### PR TITLE
Avoid depending on Jenkins in `toString()` methods

### PR DESCRIPTION
**Pull request by Vlatombe**
In some cases, it can cause guice circular dependency issues

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by Dohbedoh** [comment]
@Vlatombe I was wondering if  maybe https://github.com/jenkinsci/kubernetes-plugin/blob/kubernetes-1.29.2/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/yaml/YamlMergeStrategy.java should be adapted to ? Though it is not currently used in [PodTemplate#toString()](https://github.com/jenkinsci/kubernetes-plugin/blob/kubernetes-1.29.2/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java#L999-L1039).

**Comment by Vlatombe** [comment]
@Dohbedoh good point, I patched it too.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/PodRetention.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static PodRetention getPodTemplateDefault() {
 
     @Override
     public String toString() {
-        return getDescriptor().getDisplayName();
+        return getClass().getSimpleName();
```
Comment (amuniz) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/PodRetention.java:48]: IIUC this is part of an ID stored on disk. Is this change going to be a breaking one on plugin update? IOW will the plugin be able to handle what's already stored after this change?

Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/PodRetention.java:48]: I can probably make it compatible.

Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/PodRetention.java:48]: f85374a

**Review by amuniz — APPROVED** [review]

**Pull request merged**